include/sys/pci.h

```diff
@@ -114,4 +114,12 @@ static inline pci_device_t *pci_device_of(device_t *device) {
   return (device->bus == DEV_BUS_PCI) ? device->instance : NULL;
 }
 
+#define DEVICE_DRIVER_GEN_PCI_PROBE(vendor__id, device__id, device_name)       \
+  static int dev_generic_probe_##device_name(device_t *dev) {                  \
+    pci_device_t *pcid = pci_device_of(dev);                                   \
+    if (pcid->vendor_id != (vendor__id) || pcid->device_id != (device__id))    \
+      return 0;                                                                \
+    return 1;                                                                  \
+  }
+
```

Comment:
Please don't define functions in that way. It is not acceptable.

`bus_generic_probe` is already in `sys/bus.h`. What is the use case for `dev_generic_probe_*`? Why it is defined here?

Please add comments.

Reply:
`DEVICE_DRIVER_GEN_PCI_PROBE` was used to generated generic probe methods that checked VENDOR_ID and DEVICE_ID fields of a pci device.

I just replaced it with handwritten code with more consistent naming conventions.